bigtable/client/rpc_backoff_policy.h

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BIGTABLE_CLIENT_RPC_BACKOFF_POLICY_H_
+#define BIGTABLE_CLIENT_RPC_BACKOFF_POLICY_H_
+
+#include <bigtable/client/version.h>
+
+#include <grpc++/grpc++.h>
+#include <chrono>
+#include <memory>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Define the interface for controlling how the Bigtable client
+ * backsoff from failed RPC operations.
+ *
+ * The C++ client for Bigtable needs to hide partial and temporary
+ * failures from the application.  However, we need to give the users
+ * enough flexibility to control how many attempts are made to reissue
+ * operations, how often these attempts are executed, and how to
+ * signal that an error has occurred.
+ *
+ * The application provides an instance of this class when the Table
+ * (or TableAdmin) object is created.  This instance serves as a
+ * prototype to create new RPCBackoffPolicy objects of the same
+ * (dynamic) type and with the same initial state.
+ */
+class RPCBackoffPolicy {
+ public:
+  virtual ~RPCBackoffPolicy() = default;
+
+  /**
+   * Return a new copy of this object.
+   *
+   * Typically implemented as
+   * @code
+   *   return std::unique_ptr<RPCRetryPolicy>(new Foo(*this));
+   * @endcode
+   */
+  virtual std::unique_ptr<RPCBackoffPolicy> clone() const = 0;
+
+  /**
+   * Update the ClientContext for the next call.
+   */
+  virtual void setup(grpc::ClientContext& context) const = 0;
```

Comment:
Where does this get called and who is supposed to call it? I might just be missing it...

Reply:
You are not missing it, I broke down a large (some would say huge) change into smaller ones, the function will be used in a later PR.